data/src/main/java/com/teamagam/gimelgimel/data/message/repository/EntityMessageDataMapper.java

```diff
@@ -0,0 +1,55 @@
+package com.teamagam.gimelgimel.data.message.repository;
+
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.GeoEntity;
+import com.teamagam.gimelgimel.domain.messages.entity.Message;
+import com.teamagam.gimelgimel.domain.messages.repository.EntityMessageMapper;
+
+import java.util.HashMap;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import rx.Observable;
+
+@Singleton
+public class EntityMessageDataMapper implements EntityMessageMapper {
+
+    private HashMap<String, String> mGeometryToMessage;
+    private HashMap<String, String> mMessageToGeometry;
+
+    @Inject
+    public EntityMessageDataMapper() {
+        mGeometryToMessage = new HashMap<>();
```

Comment:
Class name and field maps names are not synchronized.
Its called **Entity**MessageMapper yet it holds a mapping between **Geometry** to messages.